Celeste.Mod.mm/Mod/UI/TextMenuExt.cs

```diff
@@ -355,5 +357,879 @@ public class IntSlider : TextMenu.Item {
+                        }
+                    } else if (Selection < 0 || Selection > CurrentMenu.Count - 1) {
+                        Selection = Calc.Clamp(Selection, 0, CurrentMenu.Count - 1);
+                        break;
+                    }
+                }
+                while (!Current.Hoverable);
+
+                if (!Current.Hoverable) {
+                    Selection = selection;
+                }
+                if (Selection != selection && Current != null) {
+                    if (selection >= 0 && CurrentMenu[selection] != null && CurrentMenu[selection].OnLeave != null) {
+                        CurrentMenu[selection].OnLeave();
+                    }
+                    Current.OnEnter?.Invoke();
+                    if (wiggle) {
+                        Audio.Play((direction > 0) ? "event:/ui/main/rollover_down" : "event:/ui/main/rollover_up");
+                        Current.SelectWiggler.Start();
+                    }
+                }
+            }
+
+            public void RecalculateSize() {
+                TitleHeight = ActiveFont.LineHeight;
+
+                LeftColumnWidth = RightColumnWidth = _MenuHeight = 0f;
+                if (Menus.Count < 1 || CurrentMenu == null)
+                    return;
+
+                foreach (TextMenu.Item item in CurrentMenu) {
+                    if (item.IncludeWidthInMeasurement) {
+                        LeftColumnWidth = Math.Max(LeftColumnWidth, item.LeftWidth());
+                    }
+                }
+                foreach (TextMenu.Item item in CurrentMenu) {
+                    if (item.IncludeWidthInMeasurement) {
+                        RightColumnWidth = Math.Max(RightColumnWidth, item.RightWidth());
+                    }
+                }
+                foreach (TextMenu.Item item in CurrentMenu) {
+                    if (item.Visible) {
+                        _MenuHeight += item.Height() + Container.ItemSpacing;
+                    }
+                }
+                _MenuHeight -= Container.ItemSpacing;
+            }
+
+            public float GetYOffsetOf(TextMenu.Item item) {
+                if (item == null) {
+                    return 0f;
+                }
+                float offset = 0f;
+                foreach (TextMenu.Item item2 in CurrentMenu) {
+                    if (item2.Visible) {
+                        offset += item2.Height() + ItemSpacing;
+                    }
+                    if (item2 == item) {
+                        break;
+                    }
+                }
+                return offset - item.Height() * 0.5f - ItemSpacing + Container.GetYOffsetOf(this);
+            }
+
+            private void menu_Added() {
+                foreach (Tuple<string, List<TextMenu.Item>> menu in delayedAddMenus) {
+                    this.Add(menu.Item1, menu.Item2);
+                }
+                MenuIndex = InitialSelection;
+            }
+
+            private void menu_Update() {
```

Comment:
Can be merged into non-menu_ method.

Also, what happened to spacing in this method?